#! /bin/sh /usr/share/dpatch/dpatch-run
## -*- diff -*-
## real-no-history-3.1.dpatch by Rafael Laboissiere <rafael@debian.org>
##
## DP: Only save history if Vsaving_history is true.  Thanks to John Eaton
## DP: for the patch, taken from the upstream Mercurial repository.
##
## DP: # HG changeset patch
## DP: # User John W. Eaton <jwe@octave.org>
## DP: # Date 1234548184 18000
## DP: # Node ID afbfd7f4fd931c9460102aec3eb073f78839276e
## DP: # Parent  767ed8cc6634851b239e3b181fa18fd7984c2678
## DP: toplev.cc (do_octave_atexit): only save history if Vsaving_history is true

@DPATCH@
diff --git a/src/toplev.cc b/src/toplev.cc
--- a/src/toplev.cc
+++ b/src/toplev.cc
@@ -984,7 +984,8 @@
 
       octave_history_write_timestamp ();
 
-      command_history::clean_up_and_save ();
+      if (Vsaving_history)
+	command_history::clean_up_and_save ();
 
       close_files ();
 
